internal/services/v1/permissions.go

```diff
@@ -461,6 +466,8 @@ func (ps *permissionServer) LookupResources(req *v1.LookupResourcesRequest, resp
 	}
 
 	alreadyPublishedPermissionedResourceIds := map[string]struct{}{}
+	var totalCountPublished uint64
+	defer telemetry.RecordLogicalChecks(totalCountPublished)
```

Comment:
This does not work. You are capturing the value of `totalCountPublished` as zero.

Demonstration:
```go
package telemetry

import (
	"github.com/stretchr/testify/require"
	"testing"
)

func TestDoesNotWork(t *testing.T) {
	var count uint64
	defer func() {
		require.Equal(t, uint64(1), loadLogicalChecksCount())
	}()
	defer RecordLogicalChecks(count)
	count++
}

func TestDoesWork(t *testing.T) {
	var count uint64
	defer func() {
		require.Equal(t, uint64(1), loadLogicalChecksCount())
	}()
	defer func() {
		RecordLogicalChecks(count)
	}()
	count++
}
```

Reply:
Fixed